postgres/src/test/java/org/jdbi/v3/postgres/TestDuration.java

```diff
@@ -0,0 +1,145 @@
+        final Duration d = handle.createQuery("select foo from intervals where id=?")
+                .bind(0, 8)
+                .mapTo(Duration.class)
+                .findOnly();
+        assertThat(d).isEqualTo(testDuration);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidDuration() {
+        handle.createQuery("select foo from intervals where id=?")
+                .bind(0, 5)
+                .mapTo(Duration.class)
+                .findOnly();
+    }
+
+    @Test
+    public void testReadNegativeDuration() {
+        handle.insert("insert into intervals(id, foo) values(?, interval '-2 days -3 hours')", 9);
+        final Duration d = handle.createQuery("select foo from intervals where id=?")
+                .bind(0, 9)
+                .mapTo(Duration.class)
+                .findOnly();
+        assertThat(d).isEqualTo(Duration.ofDays(-2).plusHours(-3));
+    }
+
+    @Test
+    public void testWriteReadNegativeDuration() {
+        handle.insert("insert into intervals(id, foo) values(?, ?)",
+                10, Duration.ofDays(-3).plusMinutes(2));
+        final Duration d = handle.createQuery("select foo from intervals where id=?")
+                .bind(0, 10)
+                .mapTo(Duration.class)
+                .findOnly();
+        assertThat(d).isEqualTo(Duration.ofDays(-3).plusMinutes(2));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWriteDurationTooBig() {
+        handle.insert("insert into intervals(id, foo) values(?, ?)",
+                11, Duration.ofDays((long)Integer.MAX_VALUE + 1));
+    }
+
+    /**
+     * This is admittedly a test of an implementation detail, but this detail is documented in
+     * {@link DurationArgumentFactory}, so this test failing is a good signal to update the documentation there.
+     */
+    @Test(expected = ArithmeticException.class)
+    public void testWriteDurationTooSmall() {
+        handle.insert("insert into intervals(id, foo) values(?, ?)",
+                12, Duration.ofSeconds(Long.MIN_VALUE));
+    }
+
+    @Test
+    public void testTinyDuration() {
+        handle.insert("insert into intervals(id, foo) values(?, interval '13us')", 13);
+        final Duration d = handle.createQuery("select foo from intervals where id=?")
+                .bind(0, 13)
+                .mapTo(Duration.class)
+                .findOnly();
+        assertThat(d).isEqualTo(Duration.ofNanos(13_000));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testDurationTooPrecise() {
+        handle.insert("insert into intervals(id, foo) values(?, ?)",
+                14, Duration.ofNanos(100));
+    }
+
+    // We guard against reading intervals with seconds too big or too small (i.e., more extreme than Long minimum and
+    // maximum values), but it's unclear how to actually create such intervals in Postgres.
```

Comment:
Via an `insert` statement directly, I can't create an `interval` with, e.g., > 2^63 - 1 seconds.

    => select interval '9223372036854775808 seconds';
    ERROR:  interval field value out of range: "9223372036854775808 seconds"
    LINE 1: select interval '9223372036854775808 seconds';

So that won't quite work (I tried this while trying to think of a test here).

I _could_ construct one of the column mappers _directly_ and _directly_ created a `PGInterval` instance and try to pass it in with a mocked `ResultSet`, but this seemed not quite worth it either (`ResultSet` has a zillion methods to implement).  So I just opted for the comment.

Further thoughts, reactions?  Open to suggestions.